base/sort.jl

```diff
@@ -82,16 +82,80 @@ issorted(itr;
     lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward) =
     issorted(itr, ord(lt,by,rev,order))
 
-function select!(v::AbstractVector, k::Union{Int,OrdinalRange}, o::Ordering)
+function partialsort!(v::AbstractVector, k::Union{Int,OrdinalRange}, o::Ordering)
     inds = indices(v, 1)
     sort!(v, first(inds), last(inds), PartialQuickSort(k), o)
-    v[k]
+
+    if k isa Integer
```

Comment:
Perhaps using dispatch for this logic would be better than branching?

Reply:
Let's go with the current version then. I've rebased again, let's not wait for too long as there were already substantial conflicts.

Reply:
Actually, `@views` seems to do the trick. See https://github.com/JuliaLang/julia/pull/23760.